tools/shared/src/main/scala/org/scalajs/core/tools/io/VirtualFiles.scala

```diff
@@ -158,38 +154,45 @@ trait VirtualRelativeScalaJSIRFile extends VirtualScalaJSIRFile
 
 /** Base trait for virtual Scala.js IR files that are serialized as binary file.
  */
-trait VirtualSerializedScalaJSIRFile extends VirtualBinaryFile with VirtualScalaJSIRFile {
-  /** Class info of this file. */
-  override def info: ir.Infos.ClassInfo = {
-    // Overridden to read only the necessary parts
-    val stream = inputStream
-    try {
-      ir.InfoSerializers.deserialize(stream)
-    } catch {
-      case e: ir.IRVersionNotSupportedException =>
-        throw new ir.IRVersionNotSupportedException(e.version, e.supported,
-            s"Failed to deserialize info of file compiled with Scala.js ${e.version}" +
-            s" (supported: ${e.supported.mkString(", ")}): $path", e)
+trait VirtualSerializedScalaJSIRFile
+    extends VirtualBinaryFile with VirtualScalaJSIRFile {
 
-      case e: IOException =>
-        throw new IOException(s"Failed to deserialize info of $path", e)
-    } finally {
-      stream.close()
+  override def entryPointsInfo: ir.EntryPointsInfo = {
+    // Overridden to read only the necessary parts
+    enrichIOExceptions {
+      val stream = inputStream
```

Comment:
It is declared without `()`, so I cannot call it with `()`. It's been there since 0.5.0-M1, basically (or at least, `VirtualTextFile.reader`, which is similar, has been). I guess it can be argued that opening the stream itself does not have *side effects* per se, since it does not modify the world. It is clearly not *pure* though, since something else modifying the world will cause that method to behave differently (e.g., throw a `FileNotFoundException`). It comes down to the distinction whether `()` signals *side effects* or *impurity*. Since getters of a mutable var do not have `()`, it cannot be true that `()` signals impurity, so it must be side effects. And according to that logic, `inputStream` should not have `()` either.

But it's fuzzy and debatable.